### DISCUSSION TITLE
[Docker] How does docker transform `EBUSD_*` to ebusd arguments?

### DISCUSSION THREAD

**Discussion by Commifreak** [john30/ebusd#614] [General]
Hi,

I noticed, that ebusd docker image can be controlled via env's. But I did not found out: how? I just seen the usage of predefined envs inside the app running inside the container, but somehow all `EBUSD_` starting env's are converted to a `--` argument? How is that working? Docker magic?
And how does docker know `EBUSD_` has to be cut?

**Comment by john30** [comment]
this is not related to docker at all. it is ebusd interpreting the env vars. docs are [here](https://github.com/john30/ebusd/wiki/2.-Run#environment-variables) and [here](https://github.com/john30/ebusd/tree/master/contrib/docker#use-of-environment-variables)

**Comment by Commifreak** [comment]
Yep, I found that already but not the code for handling those variables :/

Gruß,
R. Kluth

**Reply by john30** [reply]
not difficult to find when looking for "env": [main.cpp:1315-1354](https://github.com/john30/ebusd/blob/master/src/ebusd/main.cpp#L1315-L1354)

**Comment by Commifreak** [comment]
I even searched for „EBUSD_“ but did not found that spot. Weird.

Thanks!

Gruß,
R. Kluth
